src/utils/handleRequestWith.ts

```diff
@@ -64,7 +64,18 @@ export const handleRequestWith = (
       req.body = parseBody(req.body, req.headers)
 
       // Set document cookies on the request.
-      req.cookies = getRequestCookies(req)
+      store.hydrate()
+      req.cookies = Object.fromEntries(
```

Comment:
Alternatively, we could have a designated method on the cookie store to either:

A. Return a `{ cookies, headers }` object already derived from the internal store values.
B. Accept a request and set those properties on it implicitly.